ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -1603,8 +1603,22 @@ let migrate_send' ~__context ~vm ~dest ~live:_ ~vdi_map ~vif_map ~vgpu_map
         XenAPI.VM.set_ha_always_run ~rpc:remote.rpc ~session_id:remote.session
           ~self:new_vm ~value:true ;
       (* Send non-database metadata *)
-      Xapi_message.send_messages ~__context ~cls:`VM ~obj_uuid:vm_uuid
+      let rpc = Helpers.make_rpc ~__context in
+      let session_id = Context.get_session_id __context in
+      (* We fetch and destroy messages via RPC calls because they are stored on the master host *)
+      let messages =
+        XenAPI.Message.get ~rpc ~session_id ~cls:`VM ~obj_uuid:vm_uuid
+          ~since:(Date.of_float 0.0)
+      in
+      Xapi_message.send_messages ~__context ~cls:`VM ~obj_uuid:vm_uuid ~messages
```

Comment:
Currently, there's no way to know whether this fails in the code, only a logline is printed. This means that the messages will be lost as the messages will be destroyed.

Since the code is not allowed to fail in any case, this probably doesn't matter much as there's no nice way to treat this

Reply:
what about preventing the destruction and creating a new message?

Reply:
what's usually done in these cases? I can't see a customer being able to resolve it so maybe a new message isn't the correct way :/  I do not like however that we're destroying data here - even if it's not that important